modules/core/src/main/java/org/apache/ignite/internal/processors/platform/client/cache/ClientCacheDataRequest.java

```diff
@@ -48,4 +54,23 @@ public int txId() {
     @Override public boolean isTransactional() {
         return super.isTransactional();
     }
+
+    /** Chain cache operation future to return response when operation is completed. */
+    protected <T> IgniteInternalFuture<ClientResponse> chainFuture(
```

Comment:
Let's move to `ClientFutureUtils` or similar - seems weird to have it in one of the request handlers.

Reply:
> seems weird to have it in one of the request handlers

Why not? It's a helper method and related only to cache data requests. This class is effectively abstract, and method is supposed to be used in children handlers. We have a lot of similar methods in code (for example, `ClientCacheRequest#cacheDescriptor`) and for me it looks absolutely normal.